Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -222,10 +222,9 @@ class DatabaseContainer: NSPersistentContainer {
                 try actions(self.writableContext)
                 FetchCache.clear()
 
+                // Refresh the state by merging persistent state and local state for avoiding optimistic locking failure
                 for object in self.writableContext.updatedObjects {
-                    if object.changedValues().isEmpty {
-                        self.writableContext.refresh(object, mergeChanges: true)
-                    }
```

Comment:
We need extensive testing first, to make sure this does not make things worse. Since this impacts pretty much the whole SDK, we need to be careful with this change

Reply:
Ran it with time profiler and it adds around 1-2% of the whole write closure.
<img width="865" alt="Screenshot 2024-09-26 at 11 00 51" src="https://github.com/user-attachments/assets/16ed388a-f66c-4a42-81ad-425ef14d7a87">